File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/SetSlotPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.inventory.item.ItemStack;
import gg.mineral.api.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public final record SetSlotPacket(byte windowId, short slot, ItemStack itemStack) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeByte(windowId);
        os.writeShort(slot);
        writeSlot(os, itemStack);
    }

    @Override
    public byte getId() {
        return 0x2F;
    }
}

File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/SetSlotPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.inventory.item.ItemStack
import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class SetSlotPacket(val windowId: Byte, val slot: Short, val itemStack: ItemStack) : Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeByte(windowId.toInt())
        os.writeShort(slot.toInt())
        writeSlot(os, itemStack)
    }

    override fun getId(): Byte {
        return 0x2F
    }
}
